src/main/java/com/google/sps/model/AuthenticatedHttpServlet.java

```diff
@@ -134,41 +176,53 @@ public void doPost(
   }
 
   /**
-   * Get a Google Credential object from the authentication cookies (idToken & accessToken) in the
-   * HTTP Request
+   * Get the email of the authenticated user. If the passed idToken is invalid, this will throw an
+   * exception.
    *
    * @param request Http Request sent from client
-   * @return Credential object with accessToken.
-   * @throws CredentialVerificationException if the credentials are invalid / not present
+   * @return email of authenticated user
+   * @throws CredentialVerificationException if the idToken is invalid / not present
    * @throws GeneralSecurityException if an issue occurs with Google's verification service
    * @throws IOException if an issue occurs with Google's verification service
    */
-  private Credential getGoogleCredential(HttpServletRequest request)
+  private String getUserEmail(HttpServletRequest request)
       throws CredentialVerificationException, GeneralSecurityException, IOException {
     Cookie idTokenCookie;
-    Cookie accessTokenCookie;
-
     try {
       idTokenCookie = ServletUtility.getCookie(request, "idToken");
     } catch (CookieParseException e) {
       throw new CredentialVerificationException("idToken is not present / cannot be parsed!", e);
     }
 
+    String idToken = idTokenCookie.getValue();
+
+    return authenticationVerifier
+        .getUserEmail(idToken)
+        .orElseThrow(
+            () ->
```

Comment:
Does it work without this line?

Reply:
Hmm - I'm not sure which line you mean but the `.orElseThrow` unwraps the optional and throws an error if the value is null. This is required for proper error handling here

Reply:
Chatted offline about this being the line `() ->` - this is required as this method takes a supplier (unfortunately)

Reply:
This makes sense, because exceptions are used in exceptional circumstances, and you don't want to incur the overhead of creating an exception object every time.

Reply:
@antmar Makes sense - interesting!